### PR TITLE
Fixing issue 2660

### PR DESCRIPTION
**Pull request by jpricket**
- We were incorrectly loading the wrong extension class

### CODE REVIEW

**Comment by jpricket** [comment]
This fixes issue #2660

File: src/Test/L0/ExtensionManagerL0.cs [hunk 1]
```diff
@@ -63,6 +63,12 @@ public void LoadsTypes()
                 AssertContains<Microsoft.VisualStudio.Services.Agent.Worker.LegacyTestResults.IResultReader>(
                     manager,
                     concreteType: typeof(Microsoft.VisualStudio.Services.Agent.Worker.LegacyTestResults.JUnitResultReader));
+                AssertContains<Microsoft.VisualStudio.Services.Agent.Worker.Maintenance.IMaintenanceServiceProvider> (
+                    manager,
+                    concreteType: typeof(Microsoft.VisualStudio.Services.Agent.Worker.Build.WorkspaceMaintenanceProvider));
+                AssertContains<Microsoft.VisualStudio.Services.Agent.Worker.Maintenance.IMaintenanceServiceProvider>(
+                    manager,
+                    concreteType: typeof(Microsoft.VisualStudio.Services.Agent.Worker.Release.ReleaseDirectoryManager));
             }
```
Comment (jtpetty) [inline src/Test/L0/ExtensionManagerL0.cs]: Are there any other of these that do not have tests?

Comment (jpricket) [reply src/Test/L0/ExtensionManagerL0.cs]: Yes, lots. I didn't want to clutter this PR up with those, but we have dozens of extensions now.

**Comment by Shahyads** [comment]
I just updated the agent to 2.164.2 (win x64) on windows server 2012. it raised an error 
```
##[debug]Evaluating condition for step: 'Checkout myproject@master to s'
##[debug]Evaluating: SucceededNode()
##[debug]Evaluating SucceededNode:
##[debug]=> True
##[debug]Result: True
##[section]Starting: Checkout myproject@master to s
==============================================================================
Task         : Get sources
Description  : Get sources from a repository. Supports Git, TfsVC, and SVN repositories.
Version      : 1.0.0
Author       : Microsoft
Help         : [More Information](https://go.microsoft.com/fwlink/?LinkId=798199)
==============================================================================
##[error]Value cannot be null.
Parameter name: Checkout (6d15af64-176c-496d-b583-fd2ae21d4df4/1.0.0)
##[debug]System.ArgumentNullException: Value cannot be null.
Parameter name: Checkout (6d15af64-176c-496d-b583-fd2ae21d4df4/1.0.0)
   at Microsoft.VisualStudio.Services.Agent.Worker.Handlers.AgentPluginHandler.RunAsync()
   at Microsoft.VisualStudio.Services.Agent.Worker.TaskRunner.RunAsync() in D:\a\1\s\src\Agent.Worker\TaskRunner.cs:line 370
   at Microsoft.VisualStudio.Services.Agent.Worker.StepsRunner.RunStepAsync(IStep step, CancellationToken jobCancellationToken) in D:\a\1\s\src\Agent.Worker\StepsRunner.cs:line 215
##[section]Finishing: Checkout myproject@master to s

```

**Comment by jtpetty** [comment]
@Shahyads - This is a known issue and is bug with 2.164.2. We are in the process of creating 2.164.3 to correct that.

**Pull request merged**